lib/media/streaming_engine.js

```diff
@@ -170,6 +171,21 @@ shaka.media.StreamingEngine = class {
     const autoReset = true;
     this.failureCallbackBackoff_ =
         new shaka.net.Backoff(failureRetryParams, autoReset);
+
+    // Allow configuring the segment prefetch in middle of the playback.
+    for (const type of this.mediaStates_.keys()) {
+      const state = this.mediaStates_.get(type);
+      if (state.segmentPrefetch) {
+        state.segmentPrefetch.resetLimit(config.segmentPrefetchLimit);
+        if (!(config.segmentPrefetchLimit > 0)) {
+          // ResetLimit is still needed in this case,
+          // to abort existing prefetch operations.
+          state.segmentPrefetch = null;
```

Comment:
If we clear the `segmentPrefetch` object away if the limit is reduced to 0, we might not need all of those if-statements that check to see if the limit is >0 inside the class itself.
Instead, you should turn those into asserts.

Reply:
changed the "if prefetchLimit < 0" statements that are inside SegmentPrefetch, to goog.asserts.assert